### ISSUE TITLE
Add the ability to decorate different interfaces

### ISSUE THREAD

**Issue by Samueljoli** [valtlfelipe/hapi-sequelizejs#20]
What do you folks think of allowing devs the ability to go ahead and choose what interface is decorated with the db.

**Comment by Nargonath** [comment]
I'm not sure I understand what you mean. Do you mean to choose which part of the server you'd like to decorate i.e decorating something else than `request` object?

**Comment by Samueljoli** [comment]
@Nargonath Yes,

Ex
```js
server.register([
    {
        register: require('hapi-sequelizejs'),
        options: [
            {
                name: 'dbname', // identifier
                models: ['./server/models/**/*.js'], // paths/globs to model files
                sequelize: new Sequelize(config, opts), // sequelize instance
                sync: true, // sync models - default false
                forceSync: false, // force sync (drops tables) - default false
                decorate  : 'server' // opt to decorate the server object rather than the request interface
            },
        ],
    },
]);
```

This was inspired by me wanting access to the DB at a server `onPreStart` extension point, which only allows you access to the `server` object

**Comment by Nargonath** [comment]
@Samueljoli Don't worry no need to apologize. :wink: 

Alright I see your use case but we would need to allow decorating multiple elements because you're more likely to still need your `request` decorated as well. I wonder if that would not be the time to look at a better solution for accessing the `db` object rather than multiplying the decorated interface. It would also come hand-in-hand with #19.

**Comment by Samueljoli** [comment]
@Nargonath Ahh thanks, seems like there is a PR for this

**Comment by valtlfelipe** [comment]
Released 1.2.0 with #19. Does this solves your problem?

**Comment by valtlfelipe** [comment]
Closed due to inactivity. Please reopen if needed.
